pymongo/collection.py

```diff
@@ -2281,6 +2283,197 @@ def index_information(
             info[index.pop("name")] = index
         return info
 
+    def list_search_indexes(
+        self,
+        name: Optional[str] = None,
+        session: Optional["ClientSession"] = None,
+        comment: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> CommandCursor[Mapping[str, Any]]:
+        """Return a cursor over search indexes for the current collection.
+
+        :Parameters:
+          - `name` (optional): If given, the name of the index to search
+            for.  Only indexes with matching index names will be returned.
+            If not given, all search indexes for the current collection
+            will be returned.
+          - `session` (optional): a :class:`~pymongo.client_session.ClientSession`.
+          - `comment` (optional): A user-provided comment to attach to this
+            command.
+
+        :Returns:
+          A :class:`~pymongo.command_cursor.CommandCursor` over the result
+          set.
+
+        .. note:: requires a MongoDB server version 7.0+ Atlas cluster.
+
+        .. versionadded:: 4.5
+        """
+        if name is None:
+            pipeline: _Pipeline = [{"$listSearchIndexes": {}}]
+        else:
+            pipeline = [{"$listSearchIndexes": {"name": name}}]
+        value = self.aggregate(pipeline, session, comment=comment, **kwargs)
+        return cast(CommandCursor[Mapping[str, Any]], value)
+
+    def create_search_index(
+        self,
+        description: Union[Mapping[str, Any], SearchIndexModel],
+        session: Optional["ClientSession"] = None,
+        comment: Any = None,
+        **kwargs: Any,
+    ) -> str:
+        """Create a single search index for the current collection.
+
+        :Parameters:
+          - `description`: The index description for the new search index.
+          - `session` (optional): a
+            :class:`~pymongo.client_session.ClientSession`.
+          - `comment` (optional): A user-provided comment to attach to this
+            command.
+          - `**kwargs` (optional): optional arguments to the createSearchIndexes
+            command (like maxTimeMS) can be passed as keyword arguments.
+
+        :Returns:
+          The name of the new search index.
+
+        .. note:: requires a MongoDB server version 7.0+ Atlas cluster.
+
+        .. versionadded:: 4.5
+        """
+        if not isinstance(description, SearchIndexModel):
+            description = SearchIndexModel(description["definition"], description.get("name"))
```

Comment:
Is this behavior tested?

Reply:
Yes, by the spec test.

Reply:
Where? The index model can be passed as either a SearchIndexModel or a dict but the spec test only covers one of those.

Reply:
I think this is my last outstanding comment. We should add some manual tests for any python specific behavior like this:
```
for model_kwargs in [dict(definition=..., name=None), etc...]:
    model = SearchIndexModel(**model_kwargs)
    coll.create_search_index(model)
    coll.create_search_index(model_kwargs)
```

Reply:
Done